### PR TITLE
CI test for RL examples: Drive and VehicleFollowing 

### PR DESCRIPTION
**Pull request by Adaickalavan**
- Added CI test for RL Drive and VehicleFollowing examples, which tests the RL training process.
- Removed `make build-all-scenarios` command and only build necessary scenarios in CI to reduce test time.
- Note: The CI test for this PR will pass after #2061 is merged.

### CODE REVIEW

File: .github/workflows/ci-base-tests-linux.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ env:
   venv_dir: .venv
 
 jobs:
-  base-tests-linux:
+  base-tests:
```
Comment (Gamenot) [inline .github/workflows/ci-base-tests-linux.yml:9]: I am not against the change but is there any reason to change this?

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Copy and pasting the git commit messages is __NOT__ enough.
 - Interest vehicles now show up in Envision.
 - Seed of `hiway-v1` env can be retrieved through a new property `seed`.
 - Added `TrafficEngineActor` to describe a scenario studio defined actor that is controlled by a traffic engine.
+- Added CI test for RL Drive and VehicleFollowing examples, which tests the RL training process.
```
Comment (Gamenot) [inline CHANGELOG.md:21]: This is not really user-facing, I would consider removing it or adding a dev section. The test can be checked through `git blame`.

File: .github/workflows/ci-base-tests-linux.yml [hunk 3]
```diff
@@ -49,5 +62,36 @@ jobs:
             --ignore=./smarts/core/tests/test_smarts_memory_growth.py \
             --ignore=./smarts/core/tests/test_env_frame_rate.py \
             --ignore=./smarts/env/tests/test_benchmark.py \
-            --ignore=./examples/tests/test_learning.py \
             -k 'not test_long_determinism'
+
+  examples-rl:
```
Comment (Gamenot) [inline .github/workflows/ci-base-tests-linux.yml:67]: How long do these two tests take to run?

File: .github/workflows/ci-base-tests-linux.yml [hunk 4]
```diff
@@ -49,5 +62,36 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          cd ${GITHUB_WORKSPACE}/examples/rl/${{matrix.tests}}
+          python3.8 -m venv ${{env.venv_dir}}
+          . ${{env.venv_dir}}/bin/activate
+          pip install --upgrade pip
+          pip install wheel==0.38.4
+          pip install -e ./../../../.[camera_obs,argoverse,test]
```
Comment (saulfield) [inline .github/workflows/ci-base-tests-linux.yml:86]:
```suggestion
          pip install -e ./../../../.[camera_obs,argoverse,test,sumo]
```

**Review by saulfield — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Adaickalavan) [reply CHANGELOG.md:21]: Reverted the addition in changelog.

File: .github/workflows/ci-base-tests-linux.yml (hunk 4 above)
Comment (Adaickalavan) [reply .github/workflows/ci-base-tests-linux.yml]: Thanks, added additional pip install option.

**Comment by Gamenot** [comment]
The test times appear to be ~18 minutes and 7 minutes. I think we should find a way to run these less frequently, maybe only against certain commit messages and master.

File: .github/workflows/ci-base-tests-linux.yml (hunk 3 above)
Comment (Adaickalavan) [reply .github/workflows/ci-base-tests-linux.yml:67]: + Currently the `examples-rl` test takes [~20 minutes](https://github.com/huawei-noah/SMARTS/actions/runs/5248833298)
+ By reducing the number of routes generated in the scenarios, the `examples-rl` test time may be reduced to [~12 minutes](https://github.com/huawei-noah/SMARTS/actions/runs/5249494423)
+ Previously the `base-tests` test took [~53 minutes](https://github.com/huawei-noah/SMARTS/actions/runs/5248804421) to run when it attempted to build all scenarios.
+ By only building the necessary scenarios, the `base-tests` now takes [~6 minutes](https://github.com/huawei-noah/SMARTS/actions/runs/5248833298).
+ For a quick benefit, we may make another PR consisting of only building the necessary scenarios which will hasten the `base-tests`, while we figure how to best do CI for `examples-rl` test.

**Review by Gamenot — APPROVED** [review]

File: .github/workflows/ci-base-tests-linux.yml (hunk 1 above)
Comment (Adaickalavan) [reply .github/workflows/ci-base-tests-linux.yml:9]: With the addition of more jobs in the same workflow, following the original naming style, each of the job would have been suffixed with `-linux`. The suffix appeared a little repetitive, given the workflow's name of `SMARTS CI Base Tests Linux` which indicates it is for Linux. In short, the change was for aesthetics. This is how the test results look like now.

Tests

**Pull request merged**